packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -959,3 +923,1325 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
+///    [Overlay].
+///  * [Positioned], which can be used to size and position the overlay child in
+///    relation to the target [Overlay]'s boundaries.
+///  * [CompositedTransformFollower], which can be used to position the overlay
+///    child in relation to the linked [CompositedTransformTarget] widget.
+class OverlayPortal extends StatefulWidget {
+  /// Creates an [OverlayPortal] that renders the widget [overlayChildBuilder]
+  /// builds on the closest [Overlay] when [OverlayPortalController.show] is
+  /// called.
+  const OverlayPortal({
+    super.key,
+    required this.controller,
+    required this.overlayChildBuilder,
+    required this.child,
+  });
+
+  /// The controller to show, hide and bring to top the overlay child.
+  final OverlayPortalController controller;
+
+  _OverlayEntryWidgetState _resolveLocation(BuildContext context) {
+    final _OverlayEntryWidgetState? entry = _RenderTheatreMarker.of(context, targetRootOverlay: false)?.overlayEntryWidgetState;
+    if (entry == null) {
+      throw FlutterError.fromParts(<DiagnosticsNode>[
+        ErrorSummary('No Overlay widget found.'),
+        ErrorDescription(
+          '$runtimeType widgets require an Overlay widget ancestor.\n'
+          'An overlay lets widgets float on top of other widget children.',
+        ),
+        ErrorHint(
+          'To introduce an Overlay widget, you can either directly '
+          'include one, or use a widget that contains an Overlay itself, '
+          'such as a Navigator, WidgetApp, MaterialApp, or CupertinoApp.',
+        ),
+        ...context.describeMissingAncestor(expectedAncestorType: Overlay),
+      ]);
+    }
+    return entry;
+  }
+
+  /// A [WidgetBuilder] used to build a widget below this widget in the tree,
+  /// that renders on the closest [Overlay].
+  ///
+  /// The said widget will only be built and shown in the closest [Overlay] once
+  /// [OverlayPortalController.show] is called on the associated [controller].
+  /// It will be painted in front of the [OverlayEntry] closest to this widget
+  /// in the widget tree (which is usually the enclosing [Route]).
+  ///
+  /// The built overlay child widget, is inserted below this widget in the
+  /// widget tree, allowing it to depend on [InheritedWidget]s above it, and
+  /// be notified when the [InheritedWidget]s change.
+  ///
+  /// Unlike [child], the built overlay child can visually extend outside the
+  /// bounds of this widget without being clipped, and receive hit-test events
+  /// outside of this widget's bounds, as long as it does not extend outside of
+  /// the [Overlay] on which it is rendered.
+  final WidgetBuilder overlayChildBuilder;
+
+  /// A widget below this widget in the tree.
+  final Widget? child;
+
+  @override
+  State<OverlayPortal> createState() => _OverlayPortalState();
+}
+
+class _TopmostLocation extends _OverlayLocation with LinkedListEntry<_TopmostLocation> {
```

Comment:
consider `_OverlayLocation` or some such? I was a bit confused when reading this because "topmost location" sounded like it meant "the thing on top of everything else" and then i was confused that it was a class name.

Reply:
`_OverlayEntryLocation`?